### PR TITLE
Interop fixes

### PR DESCRIPTION
**Pull request by franz**
continuing PR #68

### CODE REVIEW

File: cmake/FindMKL.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Build LLVM-PASSES 
+set(MKL_DIR "" CACHE PATH "Path to MKL install part of oneapi")
+if(MKL_DIR STREQUAL "") 
+  message(FATAL_ERROR "icpx was found, but -DMKL_DIR option was not passed to CMake.")
+endif()
+
+# TODO: find MKL based on paths relative to where icpx lives
```
Comment (pvelesko) [inline cmake/FindMKL.cmake:7]: This should be fleshed out. MKL can always be found relative to icpx from my experience.

File: samples/CMakeLists.txt [hunk 2]
```diff
@@ -109,6 +110,13 @@ if (ICPX_EXECUTABLE)
     sycl_hip_interop
     hip_sycl_interop
     hip_sycl_interop_no_buffers)
+
+  # the ENABLE_OMP_OFFLOAD is only required to unhide MKL::sycl in intel's MKLConfig.cmake file
+  set(ENABLE_OMP_OFFLOAD ON)
+  set(MKL_THREADING sequential)
+  include(../cmake/FindMKL.cmake)
+  find_package(MKL REQUIRED CONFIG PATHS /opt/intel/oneapi/mkl/latest/lib/cmake/mkl)
```
Comment (pvelesko) [inline samples/CMakeLists.txt:118]: Should be moved to findMKL.cmake

File: samples/hip_async_interop/BinomialOptionOpenCL.cpp [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        }
+        barrier(CLK_LOCAL_MEM_FENCE);
+    }
+
+    // write result for this block to global mem
+    if(tid == 0) out[bid] = callA[0];
+
+}
+
+)";
```
Comment (pvelesko) [inline samples/hip_async_interop/BinomialOptionOpenCL.cpp:134]: Why not just load `kernel.cl` ?

File: samples/hip_async_interop/kernel.h [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+  0x0e, 0x00, 0x00, 0x00, 0xaf, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x11, 0x00, 0x00, 0x00, 0x37, 0x00, 0x03, 0x00, 0x0f, 0x00, 0x00, 0x00,
+  0xb0, 0x00, 0x00, 0x00, 0x37, 0x00, 0x03, 0x00, 0x10, 0x00, 0x00, 0x00,
+  0xb1, 0x00, 0x00, 0x00, 0x37, 0x00, 0x03, 0x00, 0x10, 0x00, 0x00, 0x00,
+  0xb2, 0x00, 0x00, 0x00, 0xf8, 0x00, 0x02, 0x00, 0xb3, 0x00, 0x00, 0x00,
+  0x39, 0x00, 0x07, 0x00, 0x0e, 0x00, 0x00, 0x00, 0xb4, 0x00, 0x00, 0x00,
+  0x12, 0x00, 0x00, 0x00, 0xb0, 0x00, 0x00, 0x00, 0xb1, 0x00, 0x00, 0x00,
+  0xb2, 0x00, 0x00, 0x00, 0xfd, 0x00, 0x01, 0x00, 0x38, 0x00, 0x01, 0x00
+};
+unsigned int KernelSpirVLength = 5796;
```
Comment (pvelesko) [inline samples/hip_async_interop/kernel.h:486]: Again, seems excessive when you could just have a SPIR-V file and use it for both OpenCL and Level0

Comment (pvelesko) [reply samples/hip_async_interop/kernel.h:486]: Seemes like you already have a `kernel.spv`?

File: src/CHIPBindings.cc [hunk 5]
```diff
@@ -1305,7 +1306,6 @@ hipError_t hipStreamWaitEvent(hipStream_t Stream, hipEvent_t Event,
                               unsigned int Flags) {
   CHIP_TRY
   CHIPInitialize();
-  NULLCHECK(Event);
```
Comment (pvelesko) [inline src/CHIPBindings.cc:1308]: Why remove this check?

**Review by pvelesko — CHANGES_REQUESTED** [review]

File: samples/CMakeLists.txt (hunk 2 above)
Comment (franz) [reply samples/CMakeLists.txt]: moved it into `cmake/mkl_and_icpx.cmake` which looks for both MKL and icpx.

File: src/CHIPBindings.cc (hunk 5 above)
Comment (franz) [reply src/CHIPBindings.cc:1308]: there is another nullcheck on event, a few lines below

**Review by pvelesko — APPROVED** [review]

**Pull request merged**